tests/test_engine.py

```diff
@@ -75,6 +84,33 @@ def test_has_table(self):
         self.successResultOf(d)
         d = engine.has_table('mytable')
         assert self.successResultOf(d) is True
+        engine.execute("DROP TABLE mytable")
+
+    def test_more_queries_than_queuepool_slots(self):
+        table = Table('my_pool_table', sqlalchemy.MetaData(),
+                      Column('id', Integer, primary_key=True),
+                      Column('name', String, nullable=False))
+        data = [{'id': 0,
+                 'name': 'a'},
+                {'id': 1,
+                 'name': 'b'},
+                {'id': 2,
+                 'name': 'c'}]
+        # max_overflow kwarg broken, but defaults to 10
+        engine = create_engine(pool_size=5)
+        engine.execute(CreateTable(table))
+        engine.execute(table.insert(), data)
+        for _ in range(25):
```

Comment:
Trying to track this down. It should be (engine.pool.size + engine.pool.max_overflow + 1); max_overflow value currently lost in the shuffle... it defaults to 10, but I am investigating introspection.